python/vineyard/cli.py

```diff
@@ -221,14 +292,29 @@ def query(client, args):
         print(f'{args.metric}: {getattr(value, args.metric)}')
 
 
-def delete_obj(client, args):
+def delete_object(client, args):
     """Utility to delete a vineyard object based on object_id."""
-    try:
-        client.delete(object_id=as_object_id(args.object_id), force=args.force, deep=args.deep)
-        print(f'The vineyard object({args.object_id}) was deleted successfully')
-    except BaseException as exc:
-        raise Exception(('The following error was encountered while deleting ' +
-                         f'the vineyard object({args.object_id}):\n{exc}')) from exc
+    if args.object_id is not None:
+        try:
+            client.delete(object_id=as_object_id(args.object_id), force=args.force, deep=args.deep)
+            print(f'The vineyard object({args.object_id}) was deleted successfully')
+        except BaseException as exc:
+            raise Exception(('The following error was encountered while deleting ' +
+                             f'the vineyard object({args.object_id}):\n{exc}')) from exc
+    elif args.regex_pattern is not None:
+        try:
+            objects = client.list_objects(pattern=args.regex_pattern, regex=True, limit=5)
+        except BaseException as exc:
+            raise Exception(f'The following error was encountered while listing vineyard objects:\n{exc}') from exc
+        for obj in objects:
+            try:
+                client.delete(object_id=as_object_id(obj.id), force=args.force, deep=args.deep)
+                print(f'The vineyard object({obj.id}) was deleted successfully')
+            except BaseException as exc:
+                raise Exception(('The following error was encountered while deleting ' +
+                                 f'the vineyard object({obj.id}):\n{exc}')) from exc
```

Comment:
If you do `from exc`, then you don't need to print to include the `exc` in the Exception's body. Python will print the trace.

Reply:
You could simply do ` raise Exception(f'Error occur when deleting the vineyard object ({obj.id})') from exc`.
Same for the others.